backend/src/seeders/seeders.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@nestjs/common';
+import { AnimalService } from '../test_example/animal.service';
+import { CreateAnimalDto } from '../test_example/dto/create-animal.dto';
+import * as fs from 'fs';
+import seedData from './seed.data';
+
+@Injectable()
+export class SeederService {
+	inject: [AnimalService];
+
+	constructor(private readonly animalService: AnimalService) {}
+
+	shouldSeed(): boolean {
+		const shouldSeedFilePath = './dist/src/seeders/.hasSeeded';
+		if (fs.existsSync(shouldSeedFilePath)) {
+			return false;
+		} else {
+			fs.writeFileSync(shouldSeedFilePath, '');
+			return true;
+		}
+	}
```

Comment:
Still not 100% fan of this if/else here

maybe the constructor can hold a variable that is read from 2 functions
```ts
constructor(private readonly animalService: AnimalService) {
     const shouldSeedFilePath = './dist/src/seeders/.hasSeeded';
}

shouldSeed(): boolean {
    return fs.existsSync(this.shouldSeedFilePath);
}

doneSeeding() {
    fs.writeFileSync(shouldSeedFilePath, '');
}
```

Reply:
Nice refactor ;-)